schema_builder.hh

```diff
@@ -237,8 +237,9 @@ public:
         _raw._wait_for_sync = sync;
         return *this;
     }
-    schema_builder& with_partitioner(std::shared_ptr<dht::i_partitioner> p) {
-        _raw._partitioner = std::move(p);
+    schema_builder& with_partitioner(sstring name, unsigned shard_count, unsigned sharding_ignore_msb_bits) {
+        auto ptr = dht::make_partitioner(name, shard_count, sharding_ignore_msb_bits);
+        _raw._partitioner = std::shared_ptr<dht::i_partitioner>(ptr.release());
         return *this;
```

Comment:
Won't this create lots of partitioner objects, even though they're all the same?

Reply:
I don't understand why you use std::shared_ptr<> if you then call release.

Reply:
See next patch. They will be cached.

Reply:
Please drop the std::shared_ptr. I don't know how to review something that appears and disappears, because I can't tell if all the problems introduced by this patch are later resolved. I don't get a clear picture of the end result.

Reply:
Do you want me to drop shard_ptr from return value of make_partitioner or from _raw._partitioner?

Reply:
Both. The whole thing confuses me.

Reply:
Sure. I can do that but it will require squashing at least 3 commits into one here. If you're ok with that - I'm happy to do it.